### ISSUE TITLE
Limit of 500 files in gs_ls() ?

### ISSUE THREAD

**Issue by emilcwiredelta** [jennybc/googlesheets#189]
As of 2022-02-03, the googlesheets package has been archived on CRAN.
I'm closing all issues and pull requests prior to archiving this GitHub repo.

Learn more at:

<https://github.com/jennybc/googlesheets#readme>

**Comment by jennybc** [comment]
I'm glad the package is useful to you!

I don't explicitly limit the listing within `googlesheets`, so I suspect it is a limit from the Google side. It's awfully nice round number.

Have you done any searching about this issue? Or asked Google directly?

It is conceivable that only 500 Sheets can be listed via the Sheets API, but that other approaches don't have this cap (e.g. the Drive API).

If you learn anything interesting, please post it here. Otherwise, this issue will remind me to do research.

**Comment by miguelriverarios** [comment]
Hello!

I've recently started using google sheets to optimize the data flow at my organization (from Drive to R) and have found it generally very useful.

I am running into a similar issue, in that gs_ls returns a maximum of 500 spreadsheets, which is very limiting for me, because I have access to and am constantly shared on workbooks from people throughout my organization. In practice this means that if I need to register a sheet that I haven't accessed within about a week, I have to re-visit the sheet in the browser and make a minor update, so that it gets put to the top of the gs_ls list, so that it can then be registered.

Based on your comment above, I did a little bit of sleuthing. I am by no means an expert (and I only understand on a surface level what the googlesheets package is doing behind the scenes), but when I looked into gs_ls I noticed that the link for the api is 

`https://spreadsheets.google.com/feeds/spreadsheets/private/full` 

and when I go there i saw this bit of code near the top

`<openSearch:totalResults>500</openSearch:totalResults>`

After scanning the contents at the link I saw that it matched the output of gs_ls.

I'm not sure if doing it through he Drive API will solve the issue (and I'm not really sure how I would check that). I also noticed that v4 of the sheets api was released, so maybe that will offer some solution.

All that to say, I think I see why it's spitting out 500, but I do not have the expertise to really go any further.
